herddb-core/src/main/java/herddb/core/system/SyscolumnsTableManager.java

```diff
@@ -61,14 +62,15 @@ public SyscolumnsTableManager(TableSpaceManager parent) {
             int pos = 1;
             for (Column c : t.columns) {
                 boolean pk = t.isPrimaryKeyColumn(c.name);
+                boolean nonNullCType = pk ? true : ColumnTypes.isNotNullDataType(c.type);
                 String data_type = ColumnTypes.typeToString(c.type);
 
                 result.add(RecordSerializer.makeRecord(
                         table,
                         "table_name", t.name,
                         "column_name", c.name,
                         "ordinal_position", pos++,
-                        "is_nullable", pk ? 0 : 1,
+                        "is_nullable", (pk || nonNullCType) ? DatabaseMetaData.columnNoNulls : DatabaseMetaData.columnNullable,
```

Comment:
isn't "pl" reduntant here ?
if ("pk") nonNullCType is already true

Reply:
You are correct. I  removed it.